components/SpotifyInterface/SpotifyInterface.c

```diff
@@ -1,289 +1,246 @@
  */
-bool Spotify_TaskInit(SpotifyInterfaceHandler_t *SpotifyInterfaceHandler)
-{
-    InterfaceHandler = SpotifyInterfaceHandler;
-    InterfaceHandler->PlaybackInfo = (PlaybackInfo_t *)malloc(sizeof(PlaybackInfo_t));
-    InterfaceHandler->UserInfo = (UserInfo_t *)malloc(sizeof(UserInfo_t));
-    PrivateHandler.Status = INIT;
-    if (InterfaceHandler->ConfigAddressInSpiffs != NULL &&
-        InterfaceHandler->IsSpotifyAuthorizedSemaphore != NULL)
-    {
-        StaticTask_t *xTaskBuffer = (StaticTask_t *)malloc(sizeof(StaticTask_t));
-        StackType_t *xStack = (StackType_t *)malloc(SPOTIFY_TASK_STACK_SIZE * sizeof(StackType_t));                 // Assuming a stack size of 400 words (adjust as needed)
-        if (xTaskBuffer == NULL || xStack == NULL)
-        {
-            ESP_LOGE(TAG, "Memory allocation failed!\n");
-            free(xTaskBuffer);
-            free(xStack);
-            return false; // Exit with an error code
-        }
-        xTaskCreateStatic(
-            Spotify_MainTask,                    // Task function
-            "Spotify_MainTask",                  // Task name (for debugging)
-            SPOTIFY_TASK_STACK_SIZE,             // Stack size (in words)
-            NULL,                                // Task parameters (passed to the task function)
-            tskIDLE_PRIORITY + SPOTIFY_PRIORITY, // Task priority (adjust as needed)
-            xStack,                              // Stack buffer
-            xTaskBuffer                          // Task control block
-        );
+// void Spotify_TaskDeinit(SpotifyInterfaceHandler_t *SpotifyInterfaceHandler)
+// {
+//     if (SpotifyInterfaceHandler->PlaybackInfo != NULL)
+//     {
+//         free(SpotifyInterfaceHandler->PlaybackInfo);
+//     }
+//     if (SpotifyInterfaceHandler->UserInfo != NULL)
+//     {
+//         free(SpotifyInterfaceHandler->UserInfo);
+//     }
+//     if (PrivateHandler.SpotifyBuffer.MessageBuffer != NULL)
+//     {
+//         free(PrivateHandler.SpotifyBuffer.MessageBuffer);
+//     }
+//     if (PrivateHandler.SpotifyBuffer.SpotifyResponseReadyFlag != NULL)
+//     {
+//         vSemaphoreDelete(PrivateHandler.SpotifyBuffer.SpotifyResponseReadyFlag);
+//     }
+//     if (SendCodeFromHttpToSpotifyTask != NULL)
+//     {
+//         vQueueDelete(SendCodeFromHttpToSpotifyTask);
+//     }
+//     ESP_LOGI(TAG, "Spotify app deinitiated successfully");
+// }
 
-        // Allocate buffer and Initialize the Spotify API call
-        PrivateHandler.SpotifyBuffer.MessageBuffer = (char *)malloc(SUPER_BUF * sizeof(char));    
-        PrivateHandler.SpotifyBuffer.SpotifyResponseReadyFlag = xSemaphoreCreateBinary();    
-        SpotifyAPICallInit(&PrivateHandler.SpotifyBuffer);
+static esp_err_t HttpEventHandler(esp_http_client_event_t *evt) 
+{
+    static int totalLen = 0;
+    // Create the queue and semaphore
 
-        ESP_LOGI(TAG, "Spotify app initiated successfully");
-    }
-    else
+    switch (evt->event_id) 
     {
```

Comment:
what is this, why replace it?

Reply:
you mean the log? cause it won't init here anymore.